changelogs/fragments/2183-java_keystore_1671_dont_crash_when_password_is_changed.yml

```diff
@@ -0,0 +1,7 @@
+---
+bugfixes:
+  - "java_keystore - disable module.run_command's internal rc check to enable
+    external rc checks; force LANG, LC_ALL and LC_MESSAGES environment variables
+    to C to rely on keytool output parsing; parse keytool output for incorrect
+    password; reenable tasks with password changes in integration tests
+    (https://github.com/ansible-collections/community.general/issues/1671)."
```

Comment:
```suggestion
  - "java_keystore - make sure that error handling is more uniform always returns ``cmd``.
    Force ``LANG``, ``LC_ALL`` and ``LC_MESSAGES`` environment variables to ``C`` to rely on ``keytool`` output parsing (https://github.com/ansible-collections/community.general/pull/2183)."
breaking_changes:
  - "java_keystore - do not error out when the password changed, but instead recreate the keystore with the new password
    (https://github.com/ansible-collections/community.general/issues/1671)."
```

Reply:
well, to be sure... that's what I plan to do:
- keep `check_rc=False`
- reformat `if rc !=0:` blocks to properly return what is documented in `RETURN`
- comment/remove (?) conditional blocks leading to overwrite the keystore in case of:
  * alias mismatch (already there, but doesn't work)
  * password mismatch (new)
- keep or not keep environment update (LANG, LC_*) ? this is not required if we don't parse keytool output for alias or password mismatch.
- fix obvious pylint hints (some else-after-return, and some unused-variable to `dummy`)

@felixfontein @russoz , is that a good plan ?

Reply:
Sounds good to me! I'd keep the environment update, even though it's not required - it will make it easier to interpret error messages in roles/playbooks.

About conditional blocks leading to overwrite: feel free to do what you prefer. Both commenting out and removing is fine for me.